clients/lightrider/lightrider-entry.js

```diff
@@ -30,14 +33,56 @@ const LR_PRESETS = {
 /** @type {BufferConstructor} */
 globalThis.Buffer = Buffer;
 
+/**
+ * @param {Connection} connection
+ * @return {Promise<LH.Puppeteer.Page>}
+ */
+async function getPageFromConnection(connection) {
+  await connection.connect();
+  const {targetInfo: mainTargetInfo} =
+    await connection.sendCommand('Target.getTargetInfo', undefined);
+  const {frameTree} = await connection.sendCommand('Page.getFrameTree', undefined);
+
+  const pptrConnection = new PptrConnection(
+    mainTargetInfo.url,
+    // @ts-expect-error Hack to access the WRS transport layer.
+    connection.channel_.root_.transport_
+  );
+
+  const browser = await Browser.create(
+    pptrConnection,
+    [] /* contextIds */,
+    false /* ignoreHTTPSErrors */,
+    undefined /* defaultViewport */,
+    undefined /* process */,
+    undefined /* closeCallback */,
+    (targetInfo) => {
+      if (targetInfo.type !== 'page' && targetInfo.type !== 'iframe') {
+        return false;
+      }
+      // TODO only connect to iframes that are related to the main target. This requires refactoring in Puppeteer: https://github.com/puppeteer/puppeteer/issues/3667.
+      return targetInfo.targetId === mainTargetInfo.targetId ||
+        targetInfo.openerId === mainTargetInfo.targetId ||
```

Comment:
we can maybe reduce this condition to just `targetInfo.targetId === mainTargetInfo.targetId` in WRS. should test with our smokes